File: Codeforces/1520/A.kt
fun main() {
    val T = readLine()?.toInt() ?: 1
    for (t in 1..T) {
        val n = readLine()?.toInt() ?: 1
        val s = readLine() ?: ""
        var x = ""
        x += s[0]
        for (i in 1 until n) {
            if (s[i] != s[i - 1]) x += s[i]
        }
        var good = true
        val seen = mutableSetOf<Char>()
        for (ch in x) {
            if (seen.contains(ch)) {
                good = false
                break
            }
            seen.add(ch)
        }
        if (good) println("YES\n") else println("NO\n")
    }
}

File: Codeforces/1520/A.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n;
    string s;
    cin >> t;
    while (t--) {
        cin >> n >> s;
        string x = "";
        x += s[0];
        for (int i = 1; i < n; ++i) {
            if (s[i] != s[i - 1]) x += s[i];
        }
        bool good = true;
        set<char> seen;
        for (const auto &ch : x) {
            if (seen.count(ch) > 0) {
                good = false;
                break;
            }
            seen.insert(ch);
        }
        cout << (good ? "YES\n" : "NO\n");
    }
    return 0;
}